src/py/flwr/server/superlink/driver/driver_servicer.py

```diff
@@ -134,7 +138,15 @@ def GetRun(
         self, request: GetRunRequest, context: grpc.ServicerContext
     ) -> GetRunResponse:
         """Get run information."""
-        raise NotImplementedError
+        log(DEBUG, "DriverServicer.GetRun")
+
+        # Init state
+        state: State = self.state_factory.state()
+
+        # Retrieve run information
+        run = state.get_run(request.run_id)
+        run_proto = None if run is None else Run(**vars(run))
+        return GetRunResponse(run=run_proto)
```

Comment:
Should we set the gRPC status code depending on the outcome here?

Reply:
We can't set the gRPC status code here. The status code must be set in the `context` in the servicer, e.g.:

``` python
class ExampleServiceServicer(example_pb2_grpc.ExampleServiceServicer):
    def ExampleMethod(self, request, context):
        if request.name == "":
            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
            context.set_details('Name field cannot be empty')
            return example_pb2.ExampleResponse()

        response = example_pb2.ExampleResponse(message=f'Hello, {request.name}!')
        return response
```

Reply:
Alternatively, I set `run` to `None` if the `run_id` is invalid, which can be detected on the other end by `GetRunResponse.HasField`.

Reply:
That's better than the previous version.

I'm afraid I don't fully understand the first comment though. This is a servicer method and we have access to the `context`. So we can set the status code using `context` if the `run_id` wasn't valid just before returning from the `GetRunResponse`, similar to how we do it in `PullTaskRes` above.